test/functional/mempool_compatibility.py

```diff
@@ -57,14 +59,17 @@ def run_test(self):
         self.start_node(1)
         assert old_tx_hash in new_node.getrawmempool()
 
-        self.log.info("Add unbroadcasted tx to mempool on new node and shutdown")
-        unbroadcasted_tx_hash = new_node.sendtoaddress(recipient, 0.0001)
+        self.log.info("Add unbroadcasted tx to mempool on new node")
+        unbroadcasted_tx_hash = new_wallet.send_self_transfer(from_node=new_node)['txid']
         assert unbroadcasted_tx_hash in new_node.getrawmempool()
         mempool = new_node.getrawmempool(True)
         assert mempool[unbroadcasted_tx_hash]['unbroadcast']
-        self.stop_node(1)
 
-        self.log.info("Move mempool.dat from new to old node")
+        self.log.info("Sync the nodes and move mempool.dat from new to old node and shutdown")
+        self.start_node(0, ['-nowallet'])
+        self.connect_nodes(0, 1)
+        self.sync_blocks()
```

Comment:
Would be nice to move up the sync where the blocks are generated to not accidentally change what the test is testing and preserving previous test behavior

Reply:
Good point, that will also reduce the diff. Done